xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp

```diff
@@ -123,6 +123,19 @@ bool CActiveAEBufferPool::Create(unsigned int totaltime)
   config.sample_rate = m_format.m_sampleRate;
   config.channel_layout = CAEUtil::GetAVChannelLayout(m_format.m_channelLayout);
 
+  // discard already created buffers
+  if (m_allSamples.size() > 0 || m_freeSamples.size() > 0)
```

Comment:
this may work but the method this is done here is not correct. buffers are only deleted by pushing them to the discarded queue. why is this needed?